### PR TITLE
Use OpenTelemetry API

### LINKED ISSUE #43: Change instrumentation to use OpenTelemetry API (indirectly)
Change the Agent API to wrap OpenTelemetry API/SDK instead of wrapping the existing internal model.

This should not require any changes to existing instrumentation, but will require changes to instrumentation tests which will need to validate now against (in-memory) OpenTelemetry Exporter.

This is an intermediate step. The end goal is for instrumentation to use OpenTelemetry API directly (#12).

~~Since OpenTelemetry API/SDK does not support the continuation feature from the existing Agent API, we can first no-op that part of the Agent API in order to limit changes to the instrumentation, and then come back after and remove that part of the Agent API completely in a subsequent PR.~~ [the continuation feature was removed in #48]

### PR DESCRIPTION
**Pull request by trask**
🎉🎉🎉

Some instrumentation was commented out in this PR, to be brought back later (tracking in #50).

Closes #43

### CODE REVIEW

**Comment by trask** [comment]
I discussed with @tylerbenson, I'm going to break this PR up into smaller commits (still one PR). Unfortunately, I'm not sure if it can be broken into smaller commits that are individually compilable/testable, but at least this will make the PR easier to review (and we squash down to a single commit during merge anyways).

**Comment by trask** [comment]
Now broken up into smaller tests. Muzzle is failing due to recently released couchbase java-client 3.0.0 (not due to changes in this PR).

**Comment by tylerbenson** [comment]
@trask you can pull this over if you like: https://github.com/DataDog/dd-trace-java/pull/1170

File: dd-java-agent/instrumentation/jax-rs-annotations-2/src/main/java/datadog/trace/instrumentation/jaxrs2/JaxRsAnnotationsDecorator.java [hunk 1]
```diff
@@ -63,11 +63,10 @@ public void onJaxRsSpan(
     }
   }
 
-  private void updateParent(AgentSpan span, final String resourceName) {
+  private void updateParent(final AgentSpan span, final String resourceName) {
     if (span == null) {
       return;
     }
-    span = span.getLocalRootSpan();
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/jax-rs-annotations-2/src/main/java/datadog/trace/instrumentation/jaxrs2/JaxRsAnnotationsDecorator.java]: This is going to cause some weirdness... I couldn't think of a better solution though.  Ideas?

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/AgentTracerImpl.java [hunk 2]
```diff
@@ -283,55 +269,54 @@ public boolean equals(final Object obj) {
...
 
-  private static final class Extractor<C> implements TextMapExtract {
-    private final Map<String, String> extracted;
-
-    private Extractor(final C carrier, final Getter<C> getter) {
-      extracted = new HashMap<>();
-      for (final String key : getter.keys(carrier)) {
-        // extracted header value
-        String s = getter.get(carrier, key);
-        // in case of multiple values in the header, need to parse
```
[the remark this answers is not in the document]
Comment (trask) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/AgentTracerImpl.java]: I don't think so, but I'm not sure that it should? (e.g. Tracestate header value is comma-separated key-value pairs)

File: dd-java-agent/instrumentation/jax-rs-annotations-2/src/main/java/datadog/trace/instrumentation/jaxrs2/JaxRsAnnotationsDecorator.java (hunk 1 above)
Comment (trask) [reply dd-java-agent/instrumentation/jax-rs-annotations-2/src/main/java/datadog/trace/instrumentation/jaxrs2/JaxRsAnnotationsDecorator.java]: I couldn't think of anything better either. I just added this to #50 so we can come back to it later.

**Pull request merged**